### PR TITLE
Add specs for creating / deleting auth tokens

### PR DESCRIPTION
**Pull request by oponder**
Having this in /v4/ will help us get closer to removing all legacy from API.

### CODE REVIEW

**Comment by oponder** [comment]
These auth token routes can then become deprecated. 

The OIDC flow is quite different, it requires a redirect to start the flow, so I Imagine it would live at something like `/v4/oidc/login`

I even think `/v4/oidc/login` and `/v4/auth-tokens` can live together for a while.

File: definitions.yaml [hunk 1]
```diff
@@ -241,3 +241,26 @@ definitions:
...
+        description: Your password as a base64 encoded string
+
+  # A response to a successful auth token request
+  V4CreateAuthTokenResponse:
+    type: object
+    properties:
+      auth-token:
+        type: string
+        description: The newly created API token
+      code:
```
Comment (marians) [inline definitions.yaml]: The successful response does not need a `code`.

Error responses use the `V4GenericResponse` type which has a code.

**Comment by oponder** [comment]
@marians: Regarding the `base64` discussion, I wouldn't want to make changes here. I don't think there is an immediate security concern, and I envision this endpoint not having a very long lifetime anyways as we start to further flesh out our other authentication solutions.

**Comment by marians** [comment]
@oponder Fine with me. base64 seems good enough.

**Comment by oponder** [comment]
@marians can I get your approve, if you are ok with base64? also noticed (thanks t @tuommaki that I had put a hyphen in the json response field name for the auth token, fixed that now as well.

**Comment by oponder** [comment]
Spec has been implemented.

File: definitions.yaml [hunk 2]
```diff
@@ -504,9 +504,6 @@ definitions:
       owner:
         type: string
         description: Name of the organization owning the cluster
-      release_version:
-        type: string
-        description: The semantic version number of this cluster
```
Comment (marians) [inline definitions.yaml]: @oponder I just noticed that this PR has removed The `release_version` attribute from the V4ClusterListItem model. That doesn't seem right. Could you please provide a follow-up to fix that?

Comment (oponder) [reply definitions.yaml]: Oops, thats some mistake I must've done while merging. Fixing.

**Pull request merged**